### ISSUE TITLE
ignoreCollisions... ignored?

### ISSUE THREAD

**Issue by alpmestan** [DavHau/mach-nix#170]
Hello,

When using `mkPython` in the following way:

``` nix
let
  mach-nix = import (builtins.fetchGit {
    url = "https://github.com/DavHau/mach-nix/";
    ref = "refs/tags/3.0.2";
  }) {};
in
(mach-nix.mkPython {
    requirements = ''
      rasa[spacy]
    '';
}).override (_: { ignoreCollisions = true; })
```

I run into the following collision:

```
building '/nix/store/i6vyh6hij93z7wqnqk5qykd39vw39q3q-python3-3.8.5-env.drv'...
collision between `/nix/store/sn8ws24ws7wx1s90dh97rjy6crbl78kg-python3.8-rasa-sdk-2.0.0/lib/python3.8/site-packages/README.md' and `/nix/store/xwbc3fdjaxfqvz1w5igak4v9wxdshibb-python3.8-rasa-2.0.2/lib/python3.8/site-packages/README.md'
```

I was hoping that the `ignoreCollisions = true;` bit would be respected, therefore not erroring out because of a simple README collision, but it seemingly isn't. Am I missing something obvious? And is there any way to workaround that kind of problem with mach-nix powered environments?

**Comment by DavHau** [comment]
Since version 3.0.0 the result of mkPython is wapped in a buildEnv. The reason for this is, that I want to allow users to pass non-python packages via `packagesExtra`.
`buildEnv` is used to include these packages into the python environment.

Therefore your `override` call goes to buildEnv and not to python.

I'm not sure what would be the best fix for it. I have to think about it for a bit. If you have any idea, let me know.

**Comment by alpmestan** [comment]
I really don't have any insightful comment about how this should be made available, but now that I understand what's going on I definitely think it would be helpful to have an escape hatch here, perhaps as simple as giving users a chance to pass some extra args/attributes to the python environment making function.

In the meantime, can you see a way to somehow pass `ignoreCollisions = true;` there, with `mach-nix` as it stands now?

**Comment by alpmestan** [comment]
Hmm, looks like the result of my `mkPython` call exposes `python` and a bunch of other helpful attributes. The following does seem to be working, and giving me the desired environment:

``` nix
let
  mach-nix = import (builtins.fetchGit {
    url = "https://github.com/DavHau/mach-nix/";
    ref = "refs/tags/3.0.2";
  }) {};
  pyenv = mach-nix.mkPython {
    requirements = ''
      rasa[spacy]
    '';
  };
  py = pyenv.python;
  pypkgs = (py.withPackages (ps: [ps.rasa])).override (_: { ignoreCollisions = true; });
in
pyenv.nixpkgs.mkShell {
  name = "rasaenv";
  buildInputs = [ pypkgs ];
}
```

**Comment by DavHau** [comment]
I pushed some intermediary solution to branch `no-buildenv`. With this version the python environment is exposed via `pyEnv`.
To ignore collisions use:
 `(mkPython {}).pyEnv.override (_:{ ignoreCollisions = true; })`

...or the solution you just posted ;)

**Comment by alpmestan** [comment]
@DavHau While my snippet above works, your patch seems like a saner and more straightforward, to-the-point alternative, and exposing the environment like this shouldn't do any harm. I'll use that branch until this lands, in a form or another.

Thanks a lot!

**Comment by DavHau** [comment]
It should be fixed with 22da85eae4bdd829b772952f72bc0a269a1819d0
I stopped using buildEnv and instead override makeWrapperArgs to include the extra packages into the PATH.

We could add a general ignoreCollisions option to mach-nix. But what's currently holding me from it, is that I don't want people becoming used to setting this option. A collision indicates a bug most of the time and should be reported/fixed rather than ignored.

**Comment by alpmestan** [comment]
Thanks!
